src/Aether/Devices/Drivers/Sgp4x.cs

```diff
@@ -0,0 +1,174 @@
+        public ReadOnlySpan<byte> GetSerialNumber()
+        {
+            Span<byte> getSerialNumberCommand = stackalloc byte[2] { 0x36, 0x82 };
+            Span<byte> serialNumberWithCRC = stackalloc byte[9];
+
+            // Write get serial number command
+            _device.Write(getSerialNumberCommand);
+
+            Thread.Sleep(1);
+
+            // Read first two bytes of serial number (+ CRC)
+            Sensirion.ReadUInt16BigEndianAndCRC8(serialNumberWithCRC.Slice(0, 3));
+            
+            // Read second two bytes of serial number (+ CRC)
+            Sensirion.ReadUInt16BigEndianAndCRC8(serialNumberWithCRC.Slice(3, 3));
+            
+            // Read third two bytes of serial number (+ CRC)
+            Sensirion.ReadUInt16BigEndianAndCRC8(serialNumberWithCRC.Slice(6, 3));
+
+            // Read serial number into array excluding CRC bytes
+            byte[] serialNumber = new byte[6];
+
+            serialNumber[0] = serialNumberWithCRC[0];
+            serialNumber[1] = serialNumberWithCRC[1];
+            serialNumber[2] = serialNumberWithCRC[3];
+            serialNumber[3] = serialNumberWithCRC[4];
+            serialNumber[4] = serialNumberWithCRC[6];
+            serialNumber[5] = serialNumberWithCRC[7];
+
+            return new ReadOnlySpan<byte>(serialNumber);
+        }
+
+        /// <summary>
+        /// Runs a self test on the device.
+        /// </summary>
+        /// <returns>True if all tests passed. False if one or more tests failed.</returns>
+        public bool RunSelfTest()
+        {
+            Span<byte> runSelfTestCommand = stackalloc byte[2] { 0x28, 0x0E };
+            Span<byte> testResultWithCRC = stackalloc byte[3];
+
+            // Write run self test command
+            _device.Write(runSelfTestCommand);
+
+            Thread.Sleep(320);
+
+            // Read test result data
+            _device.Read(testResultWithCRC);
+
+            Sensirion.ReadUInt16BigEndianAndCRC8(testResultWithCRC);
+
+            // Check result status (Most significant byte, ignore least significant non-crc byte)
+            // 0xD4 = All tests passed
+            // 0x4B = One or more test failed
+            return testResultWithCRC[0] == 0xD4;
+        }
+
+        /// <summary>
+        /// Gets the raw VOC measurements from the device.
+        /// </summary>
+        /// <param name="relativeHumidityValue">The relative humidity value expressed as a percentage.</param>
+        /// <param name="temperatureValue">The temperature value expressed in degrees celsius.</param>
+        /// <returns>The raw VOC measurement value from the sensor. If an error occurred, it will be <see langword="null"/>.</returns>
+        /// <remarks>If default relative humidity and temperature values are supplied, humidity compensation will be disabled.</remarks>
+        public VolumeConcentration? GetVOCRawMeasure(ushort relativeHumidityValue = 50, short temperatureValue = 25)
```

Comment:
These params can be of types `RelativeHumidity` and `Temperature`.

Reply:
Done